### PR TITLE
feat: Added IPConfigurations and PrivateEndpoints to virtual_network_subnet

### PR DESCRIPTION
**Pull request by YihengQi**

### CODE REVIEW

**Comment by YihengQi** [comment]
Hi there, 
It seems like I need to have both TimescaleDB and PostgresDB migrations to get the tests pass. Do I need to setup TimescaleDB locally and generate the migration file? Is there another way doing it?
Thank you!

**Comment by disq** [comment]
> Hi there, It seems like I need to have both TimescaleDB and PostgresDB migrations to get the tests pass. Do I need to setup TimescaleDB locally and generate the migration file? Is there another way doing it? Thank you!

It is possible to generate the migrations for TimescaleDB in vanilla PostgreSQL due to protocol compatibility, by replacing `postgres://` with `tsdb://` in the DSN and still pointing it to your PostgreSQL installation.

**Comment by roneli** [comment]
Seems like the migrations in the tsdb have a typo as its v0.0.6 instead of v0.6.0

File: resources/services/network/virtual_networks_mock_test.go [hunk 1]
```diff
@@ -88,6 +88,8 @@ func buildNetworkVirtualNetworksMock(t *testing.T, ctrl *gomock.Controller) serv
 	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].RouteTable.DisableBgpRoutePropagation))
 	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].RouteTable.ProvisioningState))
 	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].RouteTable.ResourceGUID))
+	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].IPConfigurations))
+	require.Nil(t, faker.FakeData(&(*vn.Subnets)[0].PrivateEndpoints))
```
Comment (roneli) [inline resources/services/network/virtual_networks_mock_test.go:92]: We need to skip these fields in the faker as they are recursive and are causing the tests to fail.
```suggestion
	require.Nil(t, faker.FakeDataSkipFields(&(*vn.Subnets)[0].IPConfigurations), []string{"PrivateEndpointProperties"})
	require.Nil(t, faker.FakeDataSkipFields(&(*vn.Subnets)[0].PrivateEndpoints), []string{"Subnet"})
```

**Review by roneli — COMMENTED** [review]
Suggestion to fix the mock tests

**Pull request merged**